### PR TITLE
fix(lint-setup): #16 added lint setup

### PR DESCRIPTION
**Pull request by deeptibaghel**
Fixes https://github.com/mozilla/fxa-shared/issues/16

### CODE REVIEW

**Comment by deeptibaghel** [comment]
@vladikoff , have added eslint setup, do we need to make it run as a part of "npm test" as well ?

File: l10n/localizeTimestamp.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ module.exports = function (options) {
             }
           }
         }
-      } catch(e) {
+      } catch (e) {
```
Comment (vladikoff) [inline l10n/localizeTimestamp.js]: nice!

File: package.json [hunk 2]
```diff
@@ -24,6 +25,8 @@
   "homepage": "https://github.com/mozilla/fxa-shared#readme",
   "devDependencies": {
     "chai": "4.1.2",
+    "eslint": "^4.19.1",
+    "eslint-plugin-fxa": "^1.0.0",
```
Comment (vladikoff) [inline package.json]: @deeptibaghel could we please use the git hash here. For example : https://github.com/mozilla/fxa-content-server/blob/master/package.json#L129 (not sure if 41504c9dd30e8b52900c15b524946aa0428aef95 is latest)

File: package.json [hunk 3]
```diff
@@ -4,7 +4,8 @@
   "description": "Shared module for FxA repositories",
   "main": "index.js",
   "scripts": {
-    "test": "mocha --recursive test"
+    "test": "mocha --recursive test",
+    "lint": "eslint email/** l10n/** test/**"
```
Comment (vladikoff) [inline package.json]: is this going to miss https://github.com/mozilla/fxa-shared/blob/master/index.js ?

Can we do something like `**/*.js` or `*.js`

File: .eslintrc [hunk 4]
```diff
@@ -0,0 +1,11 @@
+plugins:
+  - fxa
```
Comment (vladikoff) [inline .eslintrc]: We need to run `npm run lint` on travis as well 

https://github.com/mozilla/fxa-shared/blob/master/.travis.yml


Example: https://gist.github.com/ronapelbaum/ef3c3459f01b002d0d719ca205401174#file-travis-yml-L6

**Comment by vladikoff** [comment]
>  do we need to make it run as a part of "npm test" as well ?

@philbooth what is your preferred way for this? After mocha tests are done? or a git hook?

**Comment by philbooth** [comment]
> what is your preferred way for this? After mocha tests are done? or a git hook?

Fwiw I don't like linting during `npm test` because sometimes you're working on stuff that you know won't pass lint yet, and you just want the tests results ASAP without filling up the console with unrelated lint errors.

So I'd be happier if `npm test` and `npm run lint` were completely separate. We could instead have an `npm run tl` or whatever for doing both, if people want that?

I'm also not a fan of linting in a hook either. There's plenty of times when I legitimately want to push non-working code to a branch. And we already have CI set-up so doing it locally is just duplicating that effort (on my hardware!) and slowing down my flow.

But I appreciate my opinions may diverge from the majority, so really do whatever you guys think is best and I'll deal with it! 😎

**Comment by vladikoff** [comment]
Thanks @philbooth ! 

@deeptibaghel let's just add it to the CI (.travis.yml), for now. Let's put it after the test command

**Comment by vladikoff** [comment]
@deeptibaghel one more thing, don't forge to add

```
Fixes https://github.com/mozilla/fxa-shared/issues/16
```

or

```
Fixes #16
```

to the commit message (first comment in the PR). I'm gonna edit the first comment now

File: .travis.yml [hunk 5]
```diff
@@ -12,3 +12,7 @@ notifications:
       - 'irc.mozilla.org#fxa-bots'
     use_notice: false
     skip_join: false
+
+script:
+    - npm run test
```
Comment (vladikoff) [inline .travis.yml]: Quick fix ;)

Comment (deeptibaghel) [reply .travis.yml]: Yup added and squashed commits :)

**Pull request merged**